### ISSUE TITLE
K8SSAND-1038 ⁃ batch_size_fail_threshold_in_kb not being taken into account then proxying though Stargate

### ISSUE THREAD

**Issue by mikekuzak** [k8ssandra/k8ssandra#1169]
Hi All,

I'm experiencing the following issue when querying Cassandra via stargate port 9042. 

```
WARN  [Native-Transport-Requests-2] 2021-11-08 11:40:27,182 BatchStatement.java:301 - Batch for [carryroll.timeserie_double_daily] is of size 18.066KiB, exceeding specified threshold of 5.000KiB by 13.066KiB.
WARN  [Native-Transport-Requests-1] 2021-11-08 11:40:27,200 BatchStatement.java:301 - Batch for [carryroll.timeserie_double_daily] is of size 18.066KiB, exceeding specified threshold of 5.000KiB by 13.066KiB.
ERROR [Native-Transport-Requests-1] 2021-11-08 11:40:27,239 BatchStatement.java:295 - Batch for [carryroll.timeserie_metadata] is of size 50.488KiB, exceeding specified threshold of 50.000KiB by 0.488KiB. (see batch_size_fail_threshold_in_kb)
```

In order to verify the current settings I've checked the cassandra.yaml on each node, looks like it's the helm chart default which is good. The config seems correct. 

```
batch_size_warn_threshold_in_kb: 64
batch_size_fail_threshold_in_kb: 640
```

At this point I'm not sure if stargate as a proxy has another setting somewhere, or if the cassandra.yaml is not being taking into account correctly ? 

Basically the config says batch_size_fail_threshold_in_kb: 640, and the logs are saying: 50 KiB
Thanks for any input.

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1038) by [Unito](https://www.unito.io)
┆Reviewer: 5fa4593062584c006b1e04d0
┆epic: Advanced Component Configuration
┆fixVersions: k8ssandra-1.4.0
┆friendlyId: K8SSAND-1038
┆priority: Medium

**Comment by jsanda** [comment]
Hi @mikekuzak 

Stargate has a system property `stargate.unsafe.cassandra_config_path` that specifies the location of a cassandra.yaml to use. We're using this in k8ssandra-operator for dev/tests. TBH, I am not sure if there is another, better way right now. We could add a chart property that would set that system property. You would need to supply a ConfigMap with your customized cassandra.yaml. Would that work?

**Comment by jsanda** [comment]
>So Stargate has a separate cassandra.yaml different from the cassandra.yaml 

Correct. In the Cassandra pod, cassandra.yaml is generated by an init container. It uses values set in the Helm chart. This is not used by Stargate. I think Stargate just uses a stock cassandra.yamla and some properties are set via env vars. 

>any type of property which would change the batch_size_fail_threshold_in_kb property would be helpful

As was mentioned in #1166 there has been discussion about how to expose and manage configs. I don't think we will have resolution soon, so I think a more expedient solution makes sense even if it is only short term. I'll open a ticket for the ConfigMap approach.

**Comment by mikekuzak** [comment]
btw .. I check the stargate container and I can't find the cassadra.yaml file anywhere.

**Comment by jsanda** [comment]
I was wrong. Looks like Stargate builds its initial configuration in memory with support for custom config via the system property. See [here](https://github.com/stargate/stargate/blob/master/persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/Cassandra40PersistenceActivator.java).
